### PR TITLE
Simplify the url base for ckeditor.js, make relative regardless of paren...

### PR DESCRIPTION
**Pull request by StephenMurphy**
...t structure so long as cf post meta is in the wp-content folder somewhere

### CODE REVIEW

File: classes/cf-meta-js.class.php [hunk 1]
```diff
@@ -31,24 +31,17 @@ function cf_meta_js_wysiwyg_scripts() {
 	if ( ( defined( 'CF_POST_META_DISABLE_CKEDITOR' ) ) && ( CF_POST_META_DISABLE_CKEDITOR == true ) ) {
 		return;
 	}
-	$filepath = dirname(realpath(dirname(__FILE__)));
-	
-	$dir = array_pop( explode( '/', $filepath ) );
```
Comment (StephenMurphy) [inline classes/cf-meta-js.class.php]: Couple edge case vulnerabilities here, technically.

File: classes/cf-meta-js.class.php [hunk 2]
```diff
@@ -31,24 +31,17 @@ function cf_meta_js_wysiwyg_scripts() {
...
-
-	// plugins_url() is smart enough to handle mu-plugins/ or plugins/
-	if (preg_match('|wp-content/(mu\-)?plugins|', $filepath)) {
```
Comment (StephenMurphy) [inline classes/cf-meta-js.class.php]: Overcomplex.

File: classes/cf-meta-js.class.php [hunk 3]
```diff
@@ -31,24 +31,17 @@ function cf_meta_js_wysiwyg_scripts() {
...
-		$url_base = trailingslashit(plugins_url(null, $filepath));
-	}
-	else if (preg_match('|wp-content/themes|', $filepath)) {
-		$url_base = trailingslashit(get_template_directory_uri()).'plugins/';
+	$filepath = dirname(__FILE__) . '/../';//;
+	if ( strpos( $filepath, '/wp-content/' ) !== false ) {
+		$url_base = substr( $filepath, strpos( $filepath, '/wp-content/' ) );
 	}
 	else {
-	// just in case we're a symink or something...
```
Comment (StephenMurphy) [inline classes/cf-meta-js.class.php]: ..sure?

File: classes/cf-meta-js.class.php [hunk 4]
```diff
@@ -31,24 +31,17 @@ function cf_meta_js_wysiwyg_scripts() {
...
-	
-	$dir = array_pop( explode( '/', $filepath ) );
-
-	// plugins_url() is smart enough to handle mu-plugins/ or plugins/
-	if (preg_match('|wp-content/(mu\-)?plugins|', $filepath)) {
-		$url_base = trailingslashit(plugins_url(null, $filepath));
-	}
-	else if (preg_match('|wp-content/themes|', $filepath)) {
-		$url_base = trailingslashit(get_template_directory_uri()).'plugins/';
+	$filepath = dirname(__FILE__) . '/../';//;
```
Comment (cfree) [inline classes/cf-meta-js.class.php]: what's going on at the end there? `//;`

Comment (cfree) [inline classes/cf-meta-js.class.php]: also, RW

File: classes/cf-meta-js.class.php [hunk 5]
```diff
@@ -31,24 +31,17 @@ function cf_meta_js_wysiwyg_scripts() {
...
 	}
 	else {
-	// just in case we're a symink or something...
+		// just in case we're a symink or something...
 		$url_base = trailingslashit(plugins_url());
 	}
 
 	// Filter to be able to bypass "cf-post-meta" folder if installed in somewhere else
-	$wysiwyg_js_file = apply_filters('cf_meta_js_wysiwyg_script_file', $dir.'/ckeditor/ckeditor.js');
+	$wysiwyg_js_file = apply_filters('cf_meta_js_wysiwyg_script_file', 'ckeditor/ckeditor.js');
```
Comment (cfree) [inline classes/cf-meta-js.class.php]: RW

Comment (StephenMurphy) [inline classes/cf-meta-js.class.php]: Mimicking the source it's in. Want to change the whole file?

**Comment by cfree** [comment]
Seems okay. +1

**Pull request merged**